### DISCUSSION TITLE
Capstone API is Complicated (I think?)

### DISCUSSION THREAD

**Discussion by robertncole1** [nss-evening-cohort-14/evening-client-side#240] [Help Me!] [answered]
I am thinking ahead to capstone to what I want to use and the API documentation for my idea is a bit complicated. Maybe I need to go back and redo some of the API projects. However, I can't seem to figure out how to sort through the documentation to pull the data that I am looking for.

Ultimately, my end goal is to have a search function of the Discogs database for a user to create their own vinyl record collection. I think this is possible with their API? But I am totally lost.

API Documentation: https://www.discogs.com/developers/
What I have googled: Youtube videos on public APIs, searched through the Discogs public discussion tickets for similar problems

Any guidance would be helpful so that I can either scrap this idea and move on or get my hands dirty going forward.

Thanks!

**Comment by drteresavasquez** [comment]
Is this not the endpoint you're looking for? https://www.discogs.com/developers/#page:database,header:database-search

Once you conduct the search and result come up, you can allow the user to save the details you need to your Firebase. I didn't see a sample response but I think postman can't get you that. Read up on authentication to make sure a single one is needed to get responses or if you need one at all.

**Reply by robertncole1** [reply]
I think I was having trouble understanding OAuth but I got the token and then was able to see the data I needed. Not sure if I feel 100% comfortable yet but I feel better. Thanks!

**Reply by drteresavasquez** [reply]
So for that in your code, you can use axios to help with sending the headers, Here is an example: https://stackoverflow.com/questions/54487260/how-can-we-send-oauth2-0-with-axios-in-react-js

I would put those tokens in the `.env` file and use it for the requests and then when you deploy put it in the environment variables on Netlify so that it stays private.

**Comment by drteresavasquez** [comment]
@robertncole1 if you get the call working through axios using the env let me know. Otherwise we can take a look in class tomorrow through what you're getting from Postman.

**Reply by robertncole1** [reply]
I am pulling down the right data that I need through postman so I feel better, but I am still confused on how to implement the API OAuth information into the env file? Doesn't the firebase data go there? And how exactly does that tie together? Do I create a separate env file for the Discogs token? I think talking through that tomorrow would be super helpful. Sorry if I seem all over the place! 🥴

**Reply by drteresavasquez** [reply]
The env file can house any secrets. Just make sure to follow the REACT_APP_YOUR_KEY_NAME convention and make the value your key and you should be good.

**Comment by drteresavasquez** [comment, marked as answer]
@robertncole1 I think we are good to close this one out. The solution is always to make your data manipulations from an API in your helper files and not in the UI.
